frontend/src/pages/Workflows/WorkflowitemHistoryTab/WorkflowHistoryTab.js

```diff
@@ -1,45 +1,79 @@
 import React from "react";
 import { connect } from "react-redux";
-
+import ExpansionPanel from "@material-ui/core/ExpansionPanel";
+import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
+import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
+import Typography from "@material-ui/core/Typography";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import SearchIcon from "@material-ui/icons/Search";
 import { toJS } from "../../../helper";
 import ScrollingHistory from "../../Common/History/ScrollingHistory";
-import { fetchNextWorkflowitemHistoryPage, resetWorkflowitemHistory } from "./actions";
-
-class WorkflowitemHistoryTab extends React.Component {
-  componentWillUnmount() {
-    this.props.resetWorkflowitemHistory();
-  }
+import HistorySearch from "../../Common/History/HistorySearch";
+import useHistoryState from "../../Common/History/historyHook";
+import strings from "../../../localizeStrings";
+import { getWorkflowitemEventTypes } from "../../Common/History/eventTypes";
+import {
+  fetchNextWorkflowitemHistoryPage,
+  fetchFirstWorkflowitemHistoryPage,
+  resetWorkflowitemHistory
+} from "./actions";
 
-  render() {
-    const {
-      nEventsTotal,
-      historyItems,
-      fetchNextWorkflowitemHistoryPage,
-      currentHistoryPage,
-      lastHistoryPage,
-      projectId,
-      subprojectId,
-      workflowitemId,
-      isLoading,
-      getUserDisplayname
-    } = this.props;
-    return (
+const WorkflowitemHistoryTab = ({
+  users,
+  nEventsTotal,
+  events,
+  fetchFirstWorkflowitemHistoryPage,
+  fetchNextWorkflowitemHistoryPage,
+  currentHistoryPage,
+  lastHistoryPage,
+  projectId,
+  subprojectId,
+  workflowitemId,
+  isLoading,
+  getUserDisplayname
+}) => {
+  const workflowitemEventTypes = getWorkflowitemEventTypes();
+  const [{ startAt, endAt, publisher, eventType }] = useHistoryState();
+  return (
+    <>
+      <ExpansionPanel data-test="search-history">
```

Comment:
this code is a duplication of the one in the history drawer. Restructure the drawer so the workflowitem can use the code too.

Reply:
done: Added new component HistoryContainer.js